goSrc/src/vmCaller/rpc/eth_service.go

```diff
@@ -0,0 +1,1034 @@
+		GetBalanceResult: x.EncodeBytes(balance.NativeToWei(acc.Balance).Bytes()),
+	}, nil
+}
+
+// EthGetBlockByHash iterates through all headers to find a matching block height for a given hash
+func (srv *EthService) EthGetBlockByHash(req *web3.EthGetBlockByHashParams) (*web3.EthGetBlockByHashResult, error) {
+	height, err := srv.getBlockHeightByHash(req.BlockHash)
+	if err != nil {
+		return nil, err
+	}
+
+	block, err := srv.getBlockInfoAtHeight(height, req.IsTransactionsIncluded)
+	if err != nil {
+		return nil, err
+	}
+
+	return &web3.EthGetBlockByHashResult{
+		GetBlockByHashResult: block,
+	}, nil
+}
+
+// EthGetBlockByNumber returns block info at the given height
+func (srv *EthService) EthGetBlockByNumber(req *web3.EthGetBlockByNumberParams) (*web3.EthGetBlockByNumberResult, error) {
+	height, err := srv.getHeightByWordOrNumber(req.BlockNumber)
+	if err != nil {
+		return nil, err
+	}
+
+	block, err := srv.getBlockInfoAtHeight(height, req.IsTransactionsIncluded)
+	if err != nil {
+		return nil, err
+	}
+
+	return &web3.EthGetBlockByNumberResult{
+		GetBlockByNumberResult: block,
+	}, nil
+}
+
+// EthGetBlockTransactionCountByHash returns the number of transactions in a block matching a given hash
+func (srv *EthService) EthGetBlockTransactionCountByHash(req *web3.EthGetBlockTransactionCountByHashParams) (*web3.EthGetBlockTransactionCountByHashResult, error) {
+	height, err := srv.getBlockHeightByHash(req.BlockHash)
+	if err != nil {
+		return nil, err
+	}
+
+	block, err := srv.getBlockHeaderAtHeight(height)
+	if err != nil {
+		return nil, err
+	}
+
+	return &web3.EthGetBlockTransactionCountByHashResult{
+		BlockTransactionCountByHash: x.EncodeNumber(uint64(block.NumTxs)),
+	}, nil
+}
+
+// EthGetBlockTransactionCountByNumber returns the number of transactions in a block matching a given height
+func (srv *EthService) EthGetBlockTransactionCountByNumber(req *web3.EthGetBlockTransactionCountByNumberParams) (*web3.EthGetBlockTransactionCountByNumberResult, error) {
+	height, err := srv.getHeightByWordOrNumber(req.BlockNumber)
+	if err != nil {
+		return nil, err
+	}
+
+	block, err := srv.getBlockHeaderAtHeight(height)
+	if err != nil {
+		return nil, err
+	}
+
+	return &web3.EthGetBlockTransactionCountByNumberResult{
+		BlockTransactionCountByHash: x.EncodeNumber(uint64(block.NumTxs)),
+	}, nil
+}
+
+// EthGetCode returns the EVM bytecode at an address
+func (srv *EthService) EthGetCode(req *web3.EthGetCodeParams) (*web3.EthGetCodeResult, error) {
+	addr, err := x.DecodeToAddress(req.Address)
+	if err != nil {
+		return nil, err
+	}
+
+	acc, err := srv.accounts.GetAccount(addr)
+	if err != nil {
+		return nil, err
+	} else if acc == nil {
+		return nil, fmt.Errorf("account not found at address %s", req.Address)
+	}
+
+	return &web3.EthGetCodeResult{
+		Bytes: x.EncodeBytes(acc.EVMCode),
+	}, nil
+}
+
+func (srv *EthService) EthGetStorageAt(req *web3.EthGetStorageAtParams) (*web3.EthGetStorageAtResult, error) {
+	// TODO
```

Comment:
add comment to the TODO, but if possible try to do what needs to be done instead of leafing the TODO comment